spec.yaml

```
@@ -590,8 +589,13 @@ definitions:
                 sharedSecret:
                   type: string
                   description: |
-                    An optional secret to be used to access the remote web app,
-                    for example in the form of a bearer token.
+                    An optional secret to be used to access the remote web app, such as
+                    a bearer token. To prevent leaking it in logs it MUST NOT appear
+                    in any URI. If a `code` is provided, then the sending host MUST
+                    accept the short-lived bearer token when serving the web app,
+                    which was already exchanged on the first WebDAV-based interaction
```

Why is this first WebDAV-based interaction mandatory?
```suggestion
                    which can be exchanged in the code flow interaction
```

---

It is not mandatory, but it is likely what would happen in a real implementation (granted that the phrasing should change). I should probably create a Sequence Diagram for this, but the idea is the Receiver first accesses the remote share (via WebDAV), and then clicks on some "Open in Remote app" for opening a given resource that is part of that share.

I see the two gestures as sequential, from a UI perspective the only way to jump to "Open in Remote app" would be if the receiver caches (e.g. in a favorite?) the reference of the remote resource.

---

You see them as sequential, because of a dependency? I don't think I understand this.

The 'open in remote app' is especially useful if the file is in an app-specific format that the receiving server cannot display (for instance a Jupyter notebook or an excel sheet or a binary measurements file, etc). So fetching the file over WebDAV makes no sense then, it would not be able to display the file anyway.

Why couldn't you go straight from 'you received a share' to 'open in remote app' and skip the fetching?

---

The dependency - and sequence - I see is purely from a UI perspective, you're right that you could go straight to "open in remote app" from the received share.

What I mean is the user would need to be presented with a list of files in the received share (which requires a WebDAV access), prior to be able to trigger the "open in remote app" action on one of those files. And even in the case of a single-file share, the user will have to have a view of the shared resource as obtained following a `PROPFIND`, and then can the action be triggered.